File: bfs.cpp
#include<bits/stdc++.h>
using namespace std;

vector<int> adj[100002];		// creating a adjacency list

bool visited[100002]={false};	// visiting array to mark all the previous visited nodes

queue<int> q;

void bfs(int node)
{
	visited[node]=true;
	q.push(node);
	while(!q.empty())
	{
		int node=q.front();
		cout << node <<  " ";
		q.pop();
		
		for(int u:adj[node])	// visiting each unvisited child of the current node
		{
			if(!visited[u])
			{
				visited[u]=true;
				q.push(u);	
			}	
		}
	}
}

int main()
{
	int n;
	cin >> n;					// total no of nodes
	
	int edges;
	cin >> edges;
	
	/*
	*	here indexing of the nodes is done 1 based.
	*/
	
	for(int i=0;i<edges;i++)
	{
		int u,v;
		cin >> u >> v;			// nodes between which edge is present
		adj[u].push_back(v);
		adj[v].push_back(u);
	}
	
	for(int i=1;i<=n;i++)
	{
		if(!visited[i])
		{
			bfs(i);	
		}	
	}
	
}

File: BFS.py
from collections import defaultdict 
  
class Graph: 
  
    # Constructor 
    def __init__(self): 
        self.graph = defaultdict(list) 
 
    def addEdge(self,u,v): 
        self.graph[u].append(v) 
  
    def BFS(self, s): 
  
        visited = [False] * (len(self.graph)) 
  
        queue = [] 
        queue.append(s) 
        visited[s] = True
  
        while queue: 
            s = queue.pop(0) 
            print (s, end = " ") 
            for i in self.graph[s]: 
                if visited[i] == False: 
                    queue.append(i) 
                    visited[i] = True
  

g = Graph() 
g.addEdge(0, 1) 
g.addEdge(0, 2) 
g.addEdge(1, 2) 
g.addEdge(2, 0) 
g.addEdge(2, 3) 
g.addEdge(3, 3) 
  
print ("BFS"
                  " (starting from vertex 1)") 
g.BFS(1) 
